src/sentry/api/serializers/models/organization_member.py

```diff
@@ -34,3 +36,40 @@ def serialize(self, obj, attrs, user):
             'dateCreated': obj.date_added,
         }
         return d
+
+
+class OrganizationMemberWithTeamsSerializer(OrganizationMemberSerializer):
+    def get_attrs(self, item_list, user):
+        attrs = super(OrganizationMemberWithTeamsSerializer,
+                      self).get_attrs(item_list, user)
+
+        member_team_map = list(OrganizationMemberTeam.objects.filter(
+            organizationmember__in=item_list).values(
+            'organizationmember', 'team'))
+
+        teams = {team.id: team for team in Team.objects.filter(
+            id__in=[item['team'] for item in member_team_map])}
+        results = defaultdict(list)
+
+        # results is a map of member id -> team_slug[]
+        for m in member_team_map:
+            results[m['organizationmember']].append(teams[m['team']].slug)
+
+        for item in item_list:
+            teams = results[item.id] if item.id in results else []
```

Comment:
i realize we talked about this and i didn't notice then, but this could just be `teams = results.get(item.id, [])`